src/utils.js

```diff
@@ -58,6 +58,33 @@ export function getRootExpression(node) {
 }
 
 /*
+ * returns the name of the reference node passed.
+ *
+ * example: var foo = document;
+ *  The node for foo will return 'document'
+ */
+export function getNodeReferenceName(context, node) {
+  var variables = context.getScope().variables;
+  var scopeVar;
+
+  variables.forEach((variable) => {
```

Comment:
What's this loop doing? Does this need to iterate over everything? Or could you short-circuit when you find something?